pkg/logs/sidecar/proxy.go

```diff
@@ -0,0 +1,250 @@
+package sidecar
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/nats-io/nats.go/jetstream"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	tcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/kubeshop/testkube/pkg/executor"
+	"github.com/kubeshop/testkube/pkg/logs"
+	"github.com/kubeshop/testkube/pkg/logs/events"
+	"github.com/kubeshop/testkube/pkg/utils"
+)
+
+var (
+	ErrStopSignalReceived = errors.New("stop signal received")
+)
+
+const (
+	pollInterval    = time.Second
+	podStartTimeout = time.Second * 60
+	logsBuffer      = 1000
+)
+
+func NewProxy(clientset *kubernetes.Clientset, podsClient tcorev1.PodInterface, logsStream logs.Stream, js jetstream.JetStream, log *zap.SugaredLogger, namespace, executionId string) *Proxy {
```

Comment:
My advice is to use `kubernetes.Interface` instead of `*kubernetes.Clientset` as then we can provide a mock implementation for testing

Reply:
done